psi4/driver/driver_cbs.py

```diff
@@ -25,34 +25,152 @@
 #
 # @END LICENSE
 #
+"""Plan, run, and assemble QC tasks to obtain composite method, basis, & options treatments.
+
+===
+CBS
+===
+
+cbs_text_parser()
+-----------------
+
+    _parse_cbs_gufunc_string()
+    --------------------------
+    * break user string into paired method and basis stages
+
+* transform user string into cbs kwargs inc'l basic cbs_metadata
+
+
+----------------------------
+CompositeComputer.__init__()
+----------------------------
+
+    _process_cbs_kwargs()
+    ---------------------
+    * transform user kwargs into trial cbs_metadata format (aka dict spec)
+
+        _validate_cbs_inputs()
+        ----------------------
+
+            _get_default_xtpl()
+            -------------------
+            * supply default xtpl fn for stage and basis conditions
+
+            _expand_bracketed_basis()
+            -------------------------
+            * parse and validate user bases
+
+        * check and supply defaults for cbs_metadata format (various calls to above two fns)
+
+* BaseComputer.__init__()
+
+    _build_cbs_compute()
+    --------------------
+
+        _expand_scheme_orders()
+        -----------------------
+        * form f_fields dict of entries for each zeta in a scheme (single NEED; entries related by nonlinear fn)
+
+        _contract_bracketed_basis()
+        ---------------------------
+        * form basis abbr. string from basis seq
+
+    * form d_fields list of stages or stage halves from NEEDs (GRAND_NEED; items related linearly to form final val)
+    * form list of entries (entry:= mtd-bas-opt) mentioned in GRAND_NEED (MODELCHEM; redundant, naive)
+    * form subset of MODELCHEM with minimal list of jobs (job:= entry on which to call QC) to satisfy CBS (JOBS; minimal, enlightened)
+    * form superset of JOBS with maximal list of entries resulting from JOBS (TROVE)
+    * return GRAND_NEED/cbsrec, JOBS/compute_list, TROVE/trove
+
+* form task_list of AtomicComputers 1:1 from JOBS/compute_list
+
+-------------------------------
+CompositeComputer.build_tasks()
+-------------------------------
+* pass
+
+---------------------------
+CompositeComputer.compute()
+---------------------------
+* compute() for each job in task list
+
+-----------------------------------
+CompositeComputer.get_psi_results()
+-----------------------------------
+
+    Computer.get_results()
+    ----------------------
+
+        Computer._prepare_results()
+        ---------------------------
+        * get_results() for each job in task list
+        * arrange atres data into e/d/g/h fields in compute_list and copy them into cbs tables
```

Comment:
Still not sure what `atres` is. Same with `TROVE` below. And also like before, is D down lower `dipole`?

Reply:
atres, d, dg now expanded or defined. `TROVE` is also defined somewhere in the file. It's the bundle of all the calcs you've got, whether you needed each piece or not (think ccsd, when the delta was btwn ccsd(t) and mp2).

Reply:
`atres` not defined or expanded here. You use this elsewhere in this file, so defining is probably best. Thanks much for the others, though.